### PR TITLE
ci(docs-infra): run cli docs examples concurrently

### PR DESCRIPTION
**Pull request by devversion**

### CODE REVIEW

**Comment by devversion** [comment]
@IgorMinar @gkalpak I brought down the non-ivy docs example tests job to 3min and the Ivy docs examples to 4min.

This will be ultimately solved by Bazel anyway, but for now this is most likely even faster than Bazel. so we have some good baseline and it helps with #28940 😁

**Review by petebacondarwin — APPROVED** [review]

File: aio/tools/examples/run-example-e2e.js [hunk 1]
```diff
@@ -111,22 +122,25 @@ function findAndRunE2eTests(filter, outputFile, shard) {
...
-                return promise.then(() => {
-                  return runE2eTestsCLI(specPath, outputFile).then(ok => {
-                    const arr = ok ? status.passed : status.failed;
-                    arr.push(specPath);
-                  });
-                });
-              }, Promise.resolve());
+            .then(async () => {
+              const specQueue = [...e2eSpecPaths.cli];
+              const ports = await findFreePort(4000, 6000, '127.0.0.1', specQueue.length);
```
Comment (gkalpak) [inline aio/tools/examples/run-example-e2e.js]: Why do we need `specQueue.length` ports? Can't we re-use the ports on each chunk (and thus only use `cliSpecsConcurrency` ports)?

File: aio/tools/examples/run-example-e2e.js [hunk 2]
```diff
@@ -275,29 +305,32 @@ function reportStatus(status, outputFile) {
...
       return {proc: null, promise};
     }
-    proc.stdout.on('data', function(data) { process.stdout.write(data.toString()); });
-    proc.stderr.on('data', function(data) { process.stdout.write(data.toString()); });
+    proc.stdout.on('data', printMessage);
+    proc.stderr.on('data', printMessage);
+
     proc.on('close', function(returnCode) {
-      console.log(`completed: ${descr} \n`);
+      printMessage(`completed: ${descr}\n`);
```
Comment (gkalpak) [inline aio/tools/examples/run-example-e2e.js]: What happened to the extra `\n` 😱

File: aio/tools/examples/run-example-e2e.js [hunk 3]
```diff
@@ -275,29 +305,32 @@ function reportStatus(status, outputFile) {
...
       // Many tasks (e.g., tsc) complete but are actually errors;
       // Confirm return code is zero.
       returnCode === 0 || ignoreClose ? resolve(0) : reject(returnCode);
     });
     proc.on('error', function(data) {
-      console.log(`completed with error: ${descr} \n`);
-      console.log(data.toString());
+      printMessage(`completed with error: ${descr}\n`);
```
Comment (gkalpak) [inline aio/tools/examples/run-example-e2e.js]: What happened to the extra `\n` 😱

**Review by gkalpak — APPROVED** [review]
Left some minor comments. LGTM otherwise 👍

File: aio/tools/examples/run-example-e2e.js [hunk 4]
```diff
@@ -111,22 +122,25 @@ function findAndRunE2eTests(filter, outputFile, shard) {
                 (promise, specPath) => {
                   return promise.then(() => {
                     const examplePath = path.dirname(specPath);
-                    return runE2eTestsSystemJS(examplePath, outputFile).then(ok => {
-                      const arr = ok ? status.passed : status.failed;
-                      arr.push(examplePath);
-                    });
+                    return runE2eTestsSystemJS(examplePath, outputFile)
+                      .then(passed => updateStatus(specPath, passed));
```
[the remark this answers is not in the document]
Comment (devversion) [reply aio/tools/examples/run-example-e2e.js]: Why `examplePath`? :grin: For consistency with the CLI specs, using the  `specPath` should be fine right?

File: aio/tools/examples/run-example-e2e.js (hunk 1 above)
Comment (devversion) [reply aio/tools/examples/run-example-e2e.js]: I tried that but had to move it up here as ports can be incorrectly shown as "free" while they are still stuck in the `TIME_WAIT` phase of previous chunks. I will add a comment.

File: aio/tools/examples/run-example-e2e.js (hunk 3 above)
Comment (devversion) [reply aio/tools/examples/run-example-e2e.js]: I wasn't too sure if it is really necessary. I can re-add if that has some value 😄

File: aio/tools/examples/run-example-e2e.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (devversion) [reply aio/tools/examples/run-example-e2e.js]: Just double-checked. Spec path is actually referring to a  file for SystemJS specs, while it isn't for the CLI specs. There is the actual root of inconsistency. let's just use `examplePath` then.

**Comment by devversion** [comment]
@gkalpak Addressed your feedback. thanks! 

<sup>Note: This PR still needs approval from `fw-core` and `fw-i18n`</sup>

**Review by AndrewKushnir — APPROVED** [review]
LGTM from i18n side 👍

**Comment by gkalpak** [comment]
@AndrewKushnir, what about [fw-core][1]? :grin:

[1]: https://github.com/orgs/angular/teams/fw-core

**Comment by devversion** [comment]
Yeah, still needs review from @angular/fw-core I think. Can someone please have a look?

**Review by kara — APPROVED** [review]
LGTM

**Comment by kara** [comment]
caretaker note: merging to master branch since patch didn't apply cleanly. @devversion is creating a new PR for patch

**Pull request closed without merging**